File: DSA/Cpp/test/dsa/lib/strings/edit_distance_test.cpp
#include <dsa/lib/strings/edit_distance.hpp>

namespace dsa::lib::strings {
namespace {

int getMinDistance(std::string &&w1, std::string &&w2) {
    auto r1 = EditDistance().minDistance(w1, w2);
    auto r2 = EditDistance().minDistance1(w1, w2);
    assert(r1 == r2 && "Results from 2 same functions are different.");

    return r1;
}
TEST(EditDistanceTest, emptyString) {
    EXPECT_EQ(0, getMinDistance("", ""));
    EXPECT_EQ(1, getMinDistance("a", ""));
    EXPECT_EQ(1, getMinDistance("", "a"));
}
TEST(EditDistanceTest, equalString) {
    EXPECT_EQ(0, getMinDistance("abc", "abc"));
}
TEST(EditDistanceTest, case1) {
    EXPECT_EQ(3, getMinDistance("horse", "ros"));
    EXPECT_EQ(5, getMinDistance("intention", "execution"));
}

}
}

File: DSA/Java/test/dsa/lib/strings/EditDistanceTest.java
package dsa.lib.strings;

import org.junit.Test;

import static org.junit.Assert.*;

public class EditDistanceTest {
    @Test
    public void emptyString() {
        assertEquals(0, getMinDistance("", ""));
        assertEquals(1, getMinDistance("a", ""));
        assertEquals(1, getMinDistance("", "a"));
    }

    private static int getMinDistance(String w1, String w2) {
        int r1 = new EditDistance().minDistance(w1, w2);
        int r2 = new EditDistance().minDistance1(w1, w2);
        assert r1 == r2 : "Results from 2 same functions are different.";

        return r1;
    }

    @Test
    public void equalString() {
        assertEquals(0, getMinDistance("abc", "abc"));
    }

    @Test
    public void case1() {
        assertEquals(3, getMinDistance("horse", "ros"));
        assertEquals(5, getMinDistance("intention", "execution"));
    }
}
